pkg/gatewayserver/io/basicstationlns/messages/util.go

```diff
@@ -56,35 +57,35 @@ func getFCtrlAsUint(fCtrl ttnpb.FCtrl) uint {
 }
 
 func getDataRateFromIndex(bandID string, index int) (ttnpb.DataRate, bool, error) {
-	band, err := band.GetByID(bandID)
+	phy, err := band.GetByID(bandID)
 	if err != nil {
 		return ttnpb.DataRate{}, false, errDataRateIndex.WithCause(err)
 	}
-	if index >= len(band.DataRates) {
+	if index < 0 || index > math.MaxInt32 {
```

Comment:
Maybe use `len(phy.DataRates)` instead of `math.MaxInt32`. Most people wouldn't understand why you're using `math.MaxInt32`.

Reply:
I can add a comment.
`len(phy.DataRates)` does not work, because length of `n` does not mean that data rate index `n-1` is valid.